pkg/entity/tag.go

```diff
@@ -0,0 +1,13 @@
+package entity
+
+import (
+	"github.com/jinzhu/gorm"
+)
+
+// Tag is a descriptive identifier given to ease searchability
+type Tag struct {
+	gorm.Model
+
+	Value string  `sql:"type:varchar(64);unique_index:idx_tag_value"`
```

Comment:
Thinking about the validation of `Value` field, I think it's probably safer to start with stricter regex pattern, as the tag may grow with inconsistent naming pattern. e.g. `eng_team_abc` vs `Eng Team EFG`. A good starting point is `util.IsSafeKey`.

In the future, we can decouple the validation of tags with loose restrictions if needed.

Reply:
@zhouzhuojie Makes perfect sense, could `util.IsSafeKey` be re-used here or should we develop an alternate method with our own requirements?

Reply:
Added validation for this in the last commit. I modified the regex from the `IsSafeKey` slightly to allow for spaces. Let me know if this is to your liking. Thanks!